src/Integration.Vsix/CFamily/FileConfigProvider.cs

```diff
@@ -0,0 +1,116 @@
+
+            if (!IsFileInSolution(projectItem))
+            {
+                analysisLogger.LogDebug($"[CFamilyHelper] Unable to retrieve the configuration for file '{analyzedFilePath}'. Check the file is part of a project in the current solution.", analyzerOptions);
+                return null;
+            }
+
+            try
+            {
+                // Note: if the C++ tools are not installed then it's likely an exception will be thrown when
+                // the framework tries to JIT-compile the TryGet method (since it won't be able to find the MS.VS.VCProjectEngine
+                // types).
+                return CFamilyHelper.FileConfig.TryGet(analysisLogger, projectItem, analyzedFilePath);
+            }
+            catch (Exception ex) when (!Microsoft.VisualStudio.ErrorHandler.IsCriticalException(ex))
+            {
+                analysisLogger.WriteLine(CFamilyStrings.ERROR_CreatingConfig, analyzedFilePath, ex);
+                return null;
+            }
+        }
+
+        private ILogger GetAnalysisLogger(CFamilyAnalyzerOptions analyzerOptions)
+        {
+            if (analyzerOptions is CFamilyAnalyzerOptions cFamilyAnalyzerOptions &&
+                cFamilyAnalyzerOptions.CreatePreCompiledHeaders)
+            {
+                // In case the requeset is coming from PCH generation, we don't log failures.
+                // This is to avoid redundant messages while navigating unsupported files.
+                return noOpLogger;
+            }
+
+            return logger;
+        }
+
+        internal static bool IsFileInSolution(ProjectItem projectItem)
+        {
+            try
+            {
+                // Issue 667:  https://github.com/SonarSource/sonarlint-visualstudio/issues/667
+                // If you open a C++ file that is not part of the current solution then
+                // VS will cruft up a temporary vcxproj so that it can provide language
+                // services for the file (e.g. syntax highlighting). This means that
+                // even though we have what looks like a valid project item, it might
+                // not actually belong to a real project.
+                var indexOfSingleFileString = projectItem?.ContainingProject?.FullName.IndexOf("SingleFileISense", StringComparison.OrdinalIgnoreCase);
+                return indexOfSingleFileString.HasValue &&
+                       indexOfSingleFileString.Value <= 0 &&
+                       projectItem.ConfigurationManager != null &&
+                       // the next line will throw if the file is not part of a solution
+                       projectItem.ConfigurationManager.ActiveConfiguration != null;
+            }
+            catch (Exception ex) when (!Microsoft.VisualStudio.ErrorHandler.IsCriticalException(ex))
+            {
+                // Suppress non-critical exceptions
+            }
+            return false;
+        }
+
+        private class NoOpLogger : ILogger
```

Comment:
CFamilyHelper actually used a logger (and the fake one) only for the file config retrieval, so it's handy that we can now hide this hacky bit.